src/Ocelot.Provider.Kubernetes/KubernetesProviderFactory.cs

```diff
@@ -1,41 +1,40 @@
-﻿using System;
-
-using KubeClient;
-
+﻿using KubeClient;
 using Microsoft.Extensions.DependencyInjection;
-
 using Ocelot.Configuration;
 using Ocelot.Logging;
 using Ocelot.ServiceDiscovery;
+using Ocelot.ServiceDiscovery.Providers;
+using System;
 
 namespace Ocelot.Provider.Kubernetes
 {
     public static class KubernetesProviderFactory
-    {
-        public static ServiceDiscoveryFinderDelegate Get = (provider, config, route) =>
-        {
-            var factory = provider.GetService<IOcelotLoggerFactory>();
-            return GetKubeProvider(provider, config, route, factory);
-        };
+    {
+        /// <summary>
+        /// String constant used for provider type definition.
+        /// </summary>
+        public const string PollKube = "PollKube";

```

Comment:
```suggestion
        public const string PollKube = nameof(PollKube);

```

Reply:
We cannot because of [PollKubernetes](../blob/develop/src/Ocelot.Provider.Kubernetes/KubernetesProviderFactory.cs#L35) class. In docs we have [PoolConsul](https://ocelot.readthedocs.io/en/latest/features/servicediscovery.html?highlight=PollConsul#consul) shorten! If we change to `PollKubernetes` then we must update the docs! Won't it be a breaking change because of docs & configs?

Reply:
IMHO it should be fine like that, PollConsul and PollKubernetes are two distinct providers, one for consul, the other for kubernetes. Maybe I didn't get your comment @raman-m. All tests are passing, I could push the changes and resolve the conversation

Reply:
@ggnaegi [wrote](#issuecomment-1723450207):
Yeah, we can keep it as is.

But let's discuss it a little bit more...
According to the docs ([Kubernetes](../blob/develop/docs/features/kubernetes.rst)) we have 2 types: `Kube` and `PollKube`. I like these short names.
We could do the following:
1. Rename the `PollKubernetes` class to `PollKube`, and provide this small change from @RaynaldM's [review](#discussion_r1328659073). No docs updating at all, the type remains `"Type": "PollKube"`.
2. Change to `public const string PollKube = nameof(PollKubernetes);` and update current [Kubernetes docs](../blob/develop/docs/features/kubernetes.rst?plain=1#L71) like `"Type": "PollKubernetes"`.

The 1st solution is more elegant, since simply renaming the classes has no consequences.
The 2nd solution requires docs updating, and it could be a breaking change for current configurations oа Ocelot apps in production.
Which solution do you like? 😋

Reply:
@raman-m Ok, fine, let's take the first solution...